### PR TITLE
feat: update Kubernetes to 1.28.0-rc.1

### PR DESCRIPTION
**Pull request by smira**
See https://github.com/kubernetes/kubernetes/releases/tag/v1.28.0-rc.1

### CODE REVIEW

File: hack/test/e2e.sh [hunk 1]
```diff
@@ -325,7 +325,7 @@ function validate_rlimit_nofile {
 
 function validate_booted_secureboot {
   ${TALOSCTL} dmesg | grep "Secure boot enabled"
-  ${TALOSCTL} get securitystate -o json 
+  ${TALOSCTL} get securitystate -o json
```
Comment (frezbo) [inline hack/test/e2e.sh:328]: we can remove this now :sweat_smile:

**Review by frezbo — APPROVED** [review]

File: hack/test/e2e.sh (hunk 1 above)
Comment (smira) [reply hack/test/e2e.sh:328]: I overall feel we should move this to the integration tests... I think we should cut our shell scripts

Comment (frezbo) [reply hack/test/e2e.sh:328]: ahh yes, and fix how we detect secureboot in integration test. I'll do in a separate PR

**Pull request merged**